### ISSUE TITLE
Neglecting to update beta and gamma params in batchnorm

### ISSUE THREAD

**Issue by jkyl** [carpedm20/DCGAN-tensorflow#113]
Hello, 

hoping this doesn't get lost in the deluge here, but I noticed your `d_vars` and `g_vars` do not contain the beta and gamma parameters from the batchnorm layers. [This resource](https://kratzert.github.io/2016/02/12/understanding-the-gradient-flow-through-the-batch-normalization-layer.html) suggests that those parameters should be updated using Adam, along with the conv and FC weights/biases. 

A simple fix would just be to give the batchnorm layers a name (similar to how you do for conv and fc layers), and grab those variables by name (excluding the moving average/variance variables) when you assign `d_vars` and `g_vars`.

All best

**Comment by ppwwyyxx** [comment]
```python
In [2]: [k.name for k in self.d_vars]
Out[2]: 
[u'discriminator/d_h0_conv/w:0',
 u'discriminator/d_h0_conv/biases:0',
 u'discriminator/d_h1_conv/w:0',
 u'discriminator/d_h1_conv/biases:0',
 u'discriminator/d_bn1/beta:0',
 u'discriminator/d_bn1/gamma:0',
 u'discriminator/d_h2_lin/Matrix:0',
 u'discriminator/d_h2_lin/bias:0',
 u'discriminator/d_bn2/beta:0',
 u'discriminator/d_bn2/gamma:0',
 u'discriminator/d_h3_lin/Matrix:0',
 u'discriminator/d_h3_lin/bias:0']
In [3]: [k.name for k in self.g_vars]                                                                                                                                [175/403]
Out[3]: 
[u'generator/g_h0_lin/Matrix:0',
 u'generator/g_h0_lin/bias:0',
 u'generator/g_bn0/beta:0',
 u'generator/g_bn0/gamma:0',
 u'generator/g_h1_lin/Matrix:0',
 u'generator/g_h1_lin/bias:0',
 u'generator/g_bn1/beta:0',
 u'generator/g_bn1/gamma:0',
 u'generator/g_h2/w:0',
 u'generator/g_h2/biases:0',
 u'generator/g_bn2/beta:0',
 u'generator/g_bn2/gamma:0',
 u'generator/g_h3/w:0',
 u'generator/g_h3/biases:0']

```

**Comment by jkyl** [comment]
ah i see, didnt notice that you had named the BN layers in __init__().
